### PR TITLE
update adam documentation

### PR DESCRIPTION
**Pull request by grrlic**
Possibly patching minor wording issue that confuses reader to the documentation of Adam L2 penalty, since there is no obvious change whatsoever.

Fixes #42843

### CODE REVIEW

**Comment by yuhaozhang** [comment]
Hi @grrlic, I noticed that you referenced by my original issue https://github.com/pytorch/pytorch/issues/42843 for this change. Thanks for the quick response on this! 

I hope you don't mind a quick follow-up: Can you explain why the current Adam implementation stills "follows the proposal in `Decoupled Weight Decay Regularization`"? My original understanding is that only the implementation of AdamW is closely related to this paper, but I might be wrong on this too.
